### PR TITLE
[Web][SIMS-#1110] UI updates for Supporting-Users - Navigation fix

### PR DESCRIPTION
**Pull request by andrepestana-aot**
Fixes for the navigation through the formio breadcrumbs and for the submission. 

- added emits for proper navigation
- refactored updateSupportingInformation to use patchCall and handleAPICustomError
- raised an error for unknown errors

### CODE REVIEW

File: sources/packages/web/src/views/supporting-user/SupportingInformation.vue [hunk 1]
```diff
@@ -158,13 +158,41 @@ export default defineComponent({
       // Disable internal submit button.
       formioUtils.disableWizardButtons(formInstance);
       formInstance.options.buttonSettings.showSubmit = false;
+      // Handle the navigation using the breadcrumbs.
+      formInstance.on(
+        "wizardPageSelected",
+        (page: WizardNavigationEvent, index: number) => {
+          isFirstPage.value = index === 0;
+          isLastPage.value = formInstance.isLastPage();
+          // Event to set isInFirstPage, current page and isInLastPage to parent
```
Comment (ann-aot) [inline sources/packages/web/src/views/supporting-user/SupportingInformation.vue:167]: for comment, we use sentence case ending with a full stop.  you can add a full stop here and line 180

File: sources/packages/web/src/views/supporting-user/SupportingInformation.vue [hunk 2]
```diff
@@ -158,13 +158,41 @@ export default defineComponent({
...
+        context.emit(
+          "pageChanged",
+          isFirstPage.value,
+          formInstance.page,
+          isLastPage.value,
+        );
       };
       formInstance.on("prevPage", prevNextNavigation);
       formInstance.on("nextPage", prevNextNavigation);
+      context.emit(
```
Comment (ann-aot) [inline sources/packages/web/src/views/supporting-user/SupportingInformation.vue:190]: do we need to emit here?

**Review by dheepak-aot — APPROVED** [review]
LGTM.👍

**Review by ann-aot — APPROVED** [review]
nice catch @andrepestana-aot 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrepestana-aot

File: sources/packages/web/src/views/supporting-user/SupportingInformation.vue [hunk 3]
```diff
@@ -158,10 +158,32 @@ export default defineComponent({
...
       formioUtils.disableWizardButtons(formInstance);
       formInstance.options.buttonSettings.showSubmit = false;
+      // Handle the navigation using the breadcrumbs.
+      formInstance.on(
+        "wizardPageSelected",
+        (page: WizardNavigationEvent, index: number) => {
+          isFirstPage.value = index === 0;
+          isLastPage.value = formInstance.isLastPage();
+          // Event to set isInFirstPage, current page and isInLastPage to parent.
+          context.emit(
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/supporting-user/SupportingInformation.vue:168]: Are these ones (lines 168 and 181) still to be removed?

Comment (andrepestana-aot) [reply sources/packages/web/src/views/supporting-user/SupportingInformation.vue:168]: Yeah, you're right. The emits are not necessary here, only set the control variables. Thanks!

**Review by andrewsignori-aot — COMMENTED** [review]
Thanks for doing the changes. Looks good 👍

**Pull request merged**